torch/csrc/jit/symbolic_variable.h

```diff
@@ -100,6 +136,9 @@ struct SymbolicVariable {
      ->i_(a("dim"), dim);
     return r;
   }
+  SymbolicVariable type_as(const SymbolicVariable rhs) const {
+    return create(aten::type_as, {*this, rhs})[0].typeLike(rhs);
```

Comment:
`typeList(rhs)` will inherit the sizes and shapes from `rhs`. Should have been `typeLikeWithScalarType(lhs, rhs->type().scalarType())`